api_client.py

```diff
@@ -421,19 +438,20 @@ def get_tenants_from_sophos(self):
                     tenant_data = self.get_partner_organization_tenants(
                         whoami_response, access_token
                     )
-                    tenant_data["access_token"] = access_token
                 else:
-                    if (
-                        self.config.tenant_id != ""
-                        and self.config.tenant_id
-                        != whoami_response["id"]
-                    ):
-                        raise Exception(
-                            "Configuration file mention tenant id not matched with whoami data tenant id"
-                        )
-                    else:
-                        tenant_data = whoami_response
-                        tenant_data["access_token"] = access_token
+                    tenant_data = {
+                        "items": [ whoami_response ],
+                        "access_token": access_token
+                    }
+
+                if (self.config.tenant_id != ""):
+                    tenants = [ tenant for tenant in tenant_data["items"] if tenant["id"] == self.config.tenant_id ]
```

Comment:
It is wasteful to iterate through all pages of all the tenants of this partner or organization, then to throw away all but one result. Instead, you could keep the old code path that fetches just the one tenant that's configured in the config.ini and all tenants if the configuration is missing.